src/Orleans/Core/GrainAttributes.cs

```diff
@@ -248,14 +248,38 @@ namespace Providers
         /// </para>
         /// </summary>
         [AttributeUsage(AttributeTargets.Class)]
-        public sealed class StorageProviderAttribute : Attribute
+        public sealed class StorageProviderAttribute : PersistenceProviderAttribute
         {
             public StorageProviderAttribute()
             {
-                    ProviderName = Runtime.Constants.DEFAULT_STORAGE_PROVIDER_NAME;
+                ProviderName = Runtime.Constants.DEFAULT_STORAGE_PROVIDER_NAME;
             }
+        }
+
+
+        /// <summary>
+        /// The [Orleans.Providers.LogViewProvider] attribute is used to specify a log view provider for grains that extend LogViewGrain&lt;T&gt;.
+        /// <para>
+        /// If present, the [Orleans.Providers.LogViewProvider] attribute overrides any [Orleans.Providers.StorageProvider] attribute,
+        /// otherwise we try to find a storage provider as for all other grains. 
+        /// </para>
+        /// </summary>
+        [AttributeUsage(AttributeTargets.Class)]
+        public sealed class LogViewProviderAttribute : PersistenceProviderAttribute
+        {
+            public LogViewProviderAttribute()
+            {
+            }
```

Comment:
Would probably be much beneficial to extend some of these explanations straight to the code for the benefit of coder readers in later times.